src/test/scala/com/microsoft/hyperspace/index/RefreshIndexTests.scala

```diff
@@ -0,0 +1,236 @@
+      "fails as expected when all source data files are deleted.") {
+    Seq(true, false).foreach { deleteDataFolder =>
+      withSQLConf(
+        IndexConstants.INDEX_LINEAGE_ENABLED -> "true",
+        IndexConstants.REFRESH_DELETE_ENABLED -> "true") {
+        SampleData.save(
+          spark,
+          nonPartitionedDataPath,
+          Seq("Date", "RGUID", "Query", "imprs", "clicks"))
+        val nonPartitionedDataDF = spark.read.parquet(nonPartitionedDataPath)
+
+        hyperspace.createIndex(nonPartitionedDataDF, indexConfig)
+
+        if (deleteDataFolder) {
+          FileUtils.delete(new Path(nonPartitionedDataPath))
+
+          val ex = intercept[AnalysisException](hyperspace.refreshIndex(indexConfig.indexName))
+          assert(ex.getMessage.contains("Path does not exist"))
+
+        } else {
+          val dataPath = new Path(nonPartitionedDataPath, "*parquet")
+          dataPath
+            .getFileSystem(new Configuration)
+            .globStatus(dataPath)
+            .foreach(p => FileUtils.delete(p.getPath))
+
+          val ex =
+            intercept[HyperspaceException](hyperspace.refreshIndex(indexConfig.indexName))
+          assert(ex.getMessage.contains("Invalid plan for creating an index."))
+        }
+        FileUtils.delete(new Path(nonPartitionedDataPath))
+        FileUtils.delete(systemPath)
+      }
+    }
+  }
+
+  test(
+    "Validate refresh index (to handle deletes from the source data) " +
+      "fails as expected when file info for an existing source data file changes.") {
+    SampleData.save(
+      spark,
+      nonPartitionedDataPath,
+      Seq("Date", "RGUID", "Query", "imprs", "clicks"))
+    val nonPartitionedDataDF = spark.read.parquet(nonPartitionedDataPath)
+
+    withSQLConf(
+      IndexConstants.INDEX_LINEAGE_ENABLED -> "true",
+      IndexConstants.REFRESH_DELETE_ENABLED -> "true") {
+      hyperspace.createIndex(nonPartitionedDataDF, indexConfig)
+
+      // Replace a source data file with a new file with same name but different properties.
+      val deletedFile = deleteDataFile(nonPartitionedDataPath)
+      FileUtils.createFile(
+        deletedFile.getFileSystem(new Configuration),
+        deletedFile,
+        "I am some random content :).")
+
+      val ex = intercept[HyperspaceException](hyperspace.refreshIndex(indexConfig.indexName))
+      assert(
+        ex.getMessage.contains("Index refresh (to handle deleted source data) aborted. " +
+          "Existing source data file info is changed"))
+    }
+  }
+
+  /**
+   * Delete one file from a given path.
+   *
+   * @param path Path to the parent folder containing data files.
+   * @param isPartitioned Is data folder partitioned or not.
+   * @return Path to the deleted file.
+   */
+  private def deleteDataFile(path: String, isPartitioned: Boolean = false): Path = {
+    val dataPath = if (isPartitioned) {
+      new Path(s"$path/Date=2018-09-03/Query=ibraco", "*parquet")
```

Comment:
Can we just do `s"$path/*/*/` as suggested https://github.com/microsoft/hyperspace/pull/142#discussion_r491097206? It's kind of weird to have to hard code the partition value here.

Reply:
yes, done.